### DISCUSSION TITLE
Welcome to comictagger Discussions!

### DISCUSSION THREAD

**Discussion by lordwelch** [comictagger/comictagger#265] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 1986yeah** [comment]
Hi,

I just found this great tool. Thanks a lot!

I just wanted to ask if it is planned to add additional online sources for tagging, e.g. www.comicguide.de for German comics or something specific for Mangas?

Also I didn't figure out yet how to tag multiple books at once (for example add the publisher or language to a whole series). So far I need to tag every single book and with a few 1000 that might take ages =D

Thanks a lot so far! Looking forward to see new features =)

Phil

**Reply by lordwelch** [reply]
If you mean manually setting metadata, currently your only option is to use the command line
```console
comictagger "comics/" -r -s -m publisher=test -t cr
```
> I just wanted to ask if it is planned to add additional online sources for tagging, e.g. [www.comicguide.de](http://www.comicguide.de/) for German comics or something specific for Mangas?

Yes, some of the initial work is being done in #341

**Reply by 1986yeah** [reply]
Thanks for your quick reply! 

Unfortunately I didn't manage to tag a whole series at once. If I use your code, first I get the error 

`comictagger: error: argument -s/--save: not allowed with argument -r/--rename`

When I remove the  "-r"-argument, comictagger will just open (Windows). 

When I remove the backslash it gives the error that "Path is not a comic archive!"

Only if I mention a specific book in the path ("comics\BOOK.cbz") it works well. 

Do I do something wrong?

Thank you for your support!

**Reply by lordwelch** [reply]
No, that lower case r that I put in `-r` was supposed to be a capital `-R`

**Reply by 1986yeah** [reply]
Great, that works well (without the backslash)! Don't know why I didn't see that in the Wiki! Thanks for your help! 

So far I'm able to tag series, publisher, language and manga (=yes). What didn't work was "blackandwhite". Do you have another tag name for that? 
Also when I try to add credits it says that "'writer' is not a valid tag name". Is it possible to also add the credits via command line (quick and easy)?

At last I wonder if it is possible to get the info for series and issue number from the file name and tag it automatically. I tried the `-f` argument, but apparently I don't know how to use it...

My try was 
`comictagger -f "comic\BOOK.cbz"`
just to see what happens. I was expecting a list or something, but only comictagger opened again. I tried different combinations but the result was always the same.

Thanks again for your help!

**Reply by harveylime** [reply]
Thanks for this tool, I am enjoying using it.  I am also having a problem with the command line syntax.  Specifically I can't get wildcards to work with ComicTagger (on Windows).  

If I have two files a.cbz and b.cbz in a directory, and type 'dir *.cbz' I get a list of the two files, as expected.  And if I use a command line instruction to change the metadata of a specific file like 'a.cbz' using ComicTagger, it also works as expected.  

But if I have *.* or *.cbz at the end instead of 'a.cbz', to edit the metadata of all the files, ComicTagger just says it can't find *.cbz.  Am I missing something about the required wildcard syntax?  Thanks in advance.

**Reply by lordwelch** [reply]
@harveylime Wildcard syntax, aka globbing, is a feature of the shell that you use. Either cmd.exe or PowerShell since you are on Windows. cmd.exe does not support wildcards. PowerShell does support wildcards but you may need to update the version that came pre-installed on your machine.

If you can't or don't want to use PowerShell, you can use the `-g` option which will make ComicTagger expand the wildcards manually.
`comictagger.exe -g *.cbz`

ComicTagger's CLI interface will only open a folder if the `-R` or `--recursive` option is given

**Comment by harveylime** [comment]
Many thanks! I hadn't heard the term globbing before, I like it.  

As with cmd.exe my version of PowerShell required `-g` to use wildcards, presumably updating it would change that.  

New question: the maximum CPU usage of ComicTagger is low, consistent with it making use of only one core.  This is a guess, searching for core and CPU doesn't show any relevant results in discussions.  Is there any way to enable multi-core usage on the Windows platform?

Thanks again for this excellent tool!

**Reply by lordwelch** [reply]
ComicTagger currently does not have any multi-core support
